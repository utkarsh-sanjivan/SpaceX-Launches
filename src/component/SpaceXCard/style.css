.spacex-card {
    max-width: 253px;
    min-height: 400px;
    background-color: #fff;
    border-radius: 5px;
}

.spacex-card-image {
    width: 88%;
    margin: 6%;
}

.spacex-card-empty-image {
    width: 88%;
    margin: 6%;
    min-height: 44%;
}

.spacex-card-content {
    width: 80%;
    margin: 10px 6%;
}

.spacex-card-name {
    color: #535896;
    font-size: 16px;
    font-weight: 800;
    font-family: roboto;
}

.spacex-card-mission-text {
    font-size: 16px;
    font-weight: 800;
}

.spacex-card-mission-list {
    font-size: 16px;
    color: blue;
    cursor: pointer;
    text-align: center;
    width: 100%;
    margin-top: 5px;
}

.spacex-card-title {
    width: 50%;
    font-size: 14px;
    font-weight: 800;
}

.spacex-card-sub-title {
    width: 50%;
    font-size: 14px;
    color: blue;
}

@media only screen and (max-width: 300px) {
    .spacex-card-title {
        font-size: 12px;
    }

    .spacex-card-sub-title {
        font-size: 12px;
    }
}

@media only screen and (min-width: 700px) and (max-width: 775px) {
    .spacex-card-title {
        font-size: 12px;
    }

    .spacex-card-sub-title {
        font-size: 12px;
    }
}

@media only screen and (min-width: 1024px) and (max-width: 1232px) {
    .spacex-card-title {
        font-size: 10px;
    }

    .spacex-card-sub-title {
        font-size: 10px;
    }

    .spacex-card-mission-list {
        font-size: 14px;
    }
}

@media only screen and (max-width: 700px) {
    .spacex-card {
        margin: auto;
    }
}