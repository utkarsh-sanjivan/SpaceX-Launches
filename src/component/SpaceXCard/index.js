import React from 'react';
import { Empty } from 'antd'
import './style.css';

const SpaceXCard = props => (
  <div className="spacex-card">
    {props.launchDetails.links.mission_patch?
      <img class="spacex-card-image" alt="Unavailable" src={props.launchDetails.links.mission_patch} />
      : <Empty 
        className="spacex-card-empty-image"
        description={'Image Unavailable'}
        image="https://gw.alipayobjects.com/zos/antfincdn/ZHrcdLPrvN/empty.svg"
        imageStyle={{ width: '100%' }}
      />
    }
    <div className="spacex-card-content">
      <div className="spacex-card-name">{props.launchDetails.mission_name} #{props.launchDetails.flight_number}</div>
      <div className="spacex-card-mission-text">Mission Ids:</div>
      <div className="spacex-card-mission-list">&#123; list Mission Ids &#125;</div>
      <div style={{ display: 'flex' }}>
        <p className="spacex-card-title">Launch Year:</p>
        <p className="spacex-card-sub-title">{props.launchDetails.launch_year}</p></div>
      <div style={{ display: 'flex' }}>
        <p className="spacex-card-title">Successful Launch:</p>
        <p className="spacex-card-sub-title">{props.launchDetails.launch_success? 'true': 'false'}</p></div>
      <div style={{ display: 'flex' }}>
        <p className="spacex-card-title">Successful Landing:</p>
        <p style={{ cursor: 'pointer' }} className="spacex-card-sub-title">&#123; launch landing &#125;</p>
      </div>
    </div>
  </div>
)

export default SpaceXCard;