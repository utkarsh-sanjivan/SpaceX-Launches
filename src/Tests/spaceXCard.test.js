import React from 'react';
import { create } from "react-test-renderer";
import SpaceXCard from './../component/SpaceXCard/index';

test('Test SpaceX Card Component', () => {
  const component = create(<SpaceXCard 
        launchDetails={{ links: {} }}
    />);
  expect(component.toJSON()).toMatchSnapshot()
});