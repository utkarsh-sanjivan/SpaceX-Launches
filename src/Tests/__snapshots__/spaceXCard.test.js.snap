// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test SpaceX Card Component 1`] = `
<div
  className="spacex-card"
>
  <div
    className="ant-empty spacex-card-empty-image"
  >
    <div
      className="ant-empty-image"
      style={
        Object {
          "width": "100%",
        }
      }
    >
      <img
        alt="Image Unavailable"
        src="https://gw.alipayobjects.com/zos/antfincdn/ZHrcdLPrvN/empty.svg"
      />
    </div>
    <p
      className="ant-empty-description"
    >
      Image Unavailable
    </p>
  </div>
  <div
    className="spacex-card-content"
  >
    <div
      className="spacex-card-name"
    >
       #
    </div>
    <div
      className="spacex-card-mission-text"
    >
      Mission Ids:
    </div>
    <div
      className="spacex-card-mission-list"
    >
      { list Mission Ids }
    </div>
    <div
      style={
        Object {
          "display": "flex",
        }
      }
    >
      <p
        className="spacex-card-title"
      >
        Launch Year:
      </p>
      <p
        className="spacex-card-sub-title"
      />
    </div>
    <div
      style={
        Object {
          "display": "flex",
        }
      }
    >
      <p
        className="spacex-card-title"
      >
        Successful Launch:
      </p>
      <p
        className="spacex-card-sub-title"
      >
        false
      </p>
    </div>
    <div
      style={
        Object {
          "display": "flex",
        }
      }
    >
      <p
        className="spacex-card-title"
      >
        Successful Landing:
      </p>
      <p
        className="spacex-card-sub-title"
        style={
          Object {
            "cursor": "pointer",
          }
        }
      >
        { launch landing }
      </p>
    </div>
  </div>
</div>
`;
